WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.kt

```diff
@@ -0,0 +1,293 @@
+        tab_layout.setupWithViewPager(view_pager)
+
+        jetpack_terms_and_conditions.text = Html.fromHtml(
+                String.format(resources.getString(R.string.jetpack_connection_terms_and_conditions), "<u>", "</u>")
+        )
+        jetpack_terms_and_conditions.setOnClickListener {
+            WPWebViewActivity.openURL(requireContext(), WPUrlUtils.buildTermsOfServiceUrl(context))
+        }
+        jetpack_faq.setOnClickListener {
+            WPWebViewActivity.openURL(requireContext(), StatsConnectJetpackActivity.FAQ_URL)
+        }
+    }
+
+    private fun buildTitles(): List<String> {
+        val result: ArrayList<String> = ArrayList(TAB_COUNT)
+        result.add(TAB_POSITION_ALL, getString(R.string.notifications_tab_title_all))
+        result.add(TAB_POSITION_UNREAD, getString(R.string.notifications_tab_title_unread_notifications))
+        result.add(TAB_POSITION_COMMENT, getString(R.string.notifications_tab_title_comments))
+        result.add(TAB_POSITION_FOLLOW, getString(R.string.notifications_tab_title_follows))
+        result.add(TAB_POSITION_LIKE, getString(R.string.notifications_tab_title_likes))
+        return result
+    }
+
+    override fun onPause() {
+        super.onPause()
+        mShouldRefreshNotifications = true
+    }
+
+    override fun onResume() {
+        super.onResume()
+        EventBus.getDefault().post(NotificationsUnseenStatus(false))
+        if (!mAccountStore.hasAccessToken()) {
+            showConnectJetpackView()
+            connect_jetpack.visibility = View.VISIBLE
+            tab_layout.visibility = View.GONE
+            view_pager.visibility = View.GONE
+        } else {
+            connect_jetpack.visibility = View.GONE
+            tab_layout.visibility = View.VISIBLE
+            view_pager.visibility = View.VISIBLE
+            if (mShouldRefreshNotifications) {
+                fetchNotesFromRemote()
+            }
+        }
+        setSelectedTab(mLastTabPosition)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putInt(KEY_LAST_TAB_POSITION, mLastTabPosition)
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun clearToolbarScrollFlags() {
+        if (toolbar_main.layoutParams is LayoutParams) {
+            val params = toolbar_main.layoutParams as LayoutParams
+            params.scrollFlags = 0
+        }
+    }
+
+    private fun fetchNotesFromRemote() {
+        if (!isAdded) {
+            return
+        }
+        if (!NetworkUtils.isNetworkAvailable(activity)) {
+            return
+        }
+        NotificationsUpdateServiceStarter.startService(activity)
+    }
+
+    val selectedSite: SiteModel?
+        get() {
+            if (activity is WPMainActivity) {
+                val mainActivity = requireActivity() as WPMainActivity
+                return mainActivity.selectedSite
+            }
+            return null
```

Comment:
I think using `(requireActivity() as? WPMainActivity)?.selectedSite` instead of this whole getter is more readable

Reply:
Fixed in 373ef82b39e0e1be9bd172dc51a50c7574c005c2